app/api/rest_api.py

```diff
@@ -80,7 +80,8 @@ def __init__(self, services):
                     chain=lambda d: self.rest_svc.update_chain_data(d),
                     operations=lambda d: self.rest_svc.create_operation(access, d),
                     schedule=lambda d: self.rest_svc.create_schedule(d),
-                    link=lambda d: self.rest_svc.apply_potential_link(Link.from_json(d))
+                    link=lambda d: self.rest_svc.apply_potential_link(Link.from_json(d)),
+                    add_adversary=lambda d: self.rest_svc.add_adversary(d)
```

Comment:
i wouldn't create a new API option for this-- instead i would use the update_operation endpoint already available (under the POST section)